### ISSUE TITLE
Refactor REST API error handling

### ISSUE THREAD

**Issue by keithmanville** [usnistgov/dioptra#486]
This feature will refactor error handling in v1 of the REST API. Currently each endpoint defines its own errors. These are often repetitive (e.g. each endpoint has a resource not found error such as `QueueNotFound` or `JobNotFound`). The refactor should instead define more general error types and provide a mechanism for setting a more specific error message via a template. The error codes should be defined in the base error and not be configurable by the user.

There should also be a base `DioptraError` that all the generic errors inherit from. This will allow the error handler to have a fallback error that can always be caught.

Raising the error may look something like this:
```python
raise ResourceNotFoundError("Queue", resource_id=queue_id)
```
With the handler returning a templated message like this:
```python
return {
    "message": f"Not Found - The requested {resource_type} does not exist",
    "id": resource_id
    }, 404
```

Tasks:
- [x] Identify common errors that should be refactored to generic
- [x] Implement generic error types whose error messages can be configured
- [x] Replace all raised errors with the newly refactored error types

### Definition of done
- [x] The individual `errors.py` files are removed from each endpoint subdirectory
- [x] BaseDioptra error and Generic errors are defined in `restapi/errors.py`
- [x] All raised exceptions in controllers and services are replaced
- [x] All tests pass
- [x] The feature is merged into dev

**Comment by keithmanville** [comment]
@hbooth sketched up what this might look like, but I know you have already done some work here, so feel free to revise.

**Comment by jkglasbrenner** [comment]
Closed by #648
